src/index.coffee

```diff
@@ -104,17 +107,19 @@ if require.extensions
       Use CoffeeScript.register() or require the coffeescript/register module to require #{ext} files.
       """
 
-CoffeeScript._compileFile = (filename, sourceMap = no, inlineMap = no) ->
+CoffeeScript._compileFile = (filename, options = {}) ->
   raw = fs.readFileSync filename, 'utf8'
   # Strip the Unicode byte order mark, if this file begins with one.
   stripped = if raw.charCodeAt(0) is 0xFEFF then raw.substring 1 else raw
 
+  Object.assign options,
```

Comment:
Is mutating `options` intentional here? Or did you mean something like `options = Object.assign {}, options, {stuff}`?

Reply:
What's the difference?

Reply:
```coffee
myOptions = {foo: true}
CoffeeScript._compileFile "some-file", myOptions
console.log myOptions.filename # "some-file"
```

That `myOptions` has changed after calling `CoffeeScript._compileFile` is a bit unexpected (this is true for any function that takes an `options` object and is a common mistake in JavaScript). Unless you _have_ to mutate `options` for stuff to work, it’s generally better not to do so to avoid bugs further down the road.

Reply:
Fair enough. I was relying on the fact that this is ostensibly a private method, and it’s only called once in our entire codebase; but it can always be safer.